### PR TITLE
Adjusted hitboxes, refactored code

### PR DESCRIPTION
**Pull request by Germlord**
-Removed unused "state" variable
-Changed "flag" variable to "flyUp" and made it boolean so that it would be more readable
-Changed "x" and "y" variables in enemies/clouds sections to "width" and "height" to be more readable and consistent
-Moved ninja hero to x=390 so that he would be better centered
-Tweaked instructions modal to be more clear and up to date
-Moved the contact area in the jumpsOn() function downward to avoid obvious hitbox discrepancies
-Removed redundant "otherBottom" variable from jumpsOn()
-Modified jumpsOn() method so that the player character does not have to have their center over the enemy. This is more of a preferential change, it just seemed that the old method was leading to a lot of "unfair" deaths.
-Cut out empty space in character pictures so that their hitboxes would function better, also modified their width, height, and y2 values so that they would still display well.

### CODE REVIEW

**Comment by EdwardDunn** [comment]
@Germlord Thanks for fixing the hit boxes issue, you found a good solution to it. I agree with all your changes, your really cleaning up the code! 

A new HUD display has been added, would you mind pulling the recent changes?

**Comment by Germlord** [comment]
No problem, I’ll get that taken care of today.

**Comment by Germlord** [comment]
@EdwardDunn okay, this should be ready to merge. Also, here is the list of the other changes that I made, since I had another bug fix I was preparing:

-Made initial startLevel() call use the currentLevel variable directly, to make for easier debugging.
-Made our check for whether the game was complete based on a direct constant, “totalLevels”, rather than the LEVEL_CLOUDS variable.
-Removed levelNumber variable and replaced it with currentLevel in most cases; also made startLevel() a parameterless function.
-Moved constants all above the non-constant variables, as it seemed like a more logical way—just in case we add a variable that might calculate off a constant.
-Renamed enemies so that the code would be more intuitively readable. Plus for style, a little bit.
-Removed unnecessary sethighscore function.
-Simplified highscore variable to an integer variable.
-Added explanation for alpha variable.
-Added if statement so that clouds won’t appear in level 5, the indoor level.
-Changed formula for cloud creation so that players can still find them while proceeding to the right.
-Tweaked end-of-level if statement so that clouds wouldn’t suddenly appear upon game-complete.

tl;dr, I fixed the clouds so that they would mostly spawn in the direction of player movement, which should fix Issue #157. I also removed clouds for the indoor level (level 5). Otherwise just did basic code cleaning.

**Comment by EdwardDunn** [comment]
@Germlord your really improving the game, thanks!

**Comment by EdwardDunn** [comment]
@Germlord Hey, I had to revert your pull request for #151 as there was an issue with the refactoring you did. The canvas is just blank, I haven't had a chance to look at what could be the cause yet. Let me know if you find the issue.

**Comment by EdwardDunn** [comment]
@Germlord The suggested fix for #151 does not resolve the issue. I have undone the revert for now, if you could have look at the issue that would be great.

**Comment by Germlord** [comment]
@EdwardDunn Thanks for letting me know. The issue should be all fixed now in the recent pull request.
